src/expr.rs

```diff
@@ -455,9 +454,19 @@ fn rewrite_closure(capture: ast::CaptureBy,
 
     // We couldn't format the closure body as a single line expression; fall
     // back to block formatting.
-    let body_rewrite = inner_block.rewrite(&context, budget, Indent::empty());
+    let body_rewrite = try_opt!(inner_block.rewrite(&context, budget, Indent::empty()));
 
-    Some(format!("{} {}", prefix, try_opt!(body_rewrite)))
+    let block_threshold = context.config.closure_block_indent_threshold;
```

Comment:
Well, block formatting is always *at least* as compact. In case of a tie, one could opt for alignment.

Your argument makes sense though. Reading large blocks of code at very deep indentation isn't optimal I agree.